From 8e797ceac7ff4d6ccbac91c689101f6a5826fff9 Mon Sep 17 00:00:00 2001
From: Yuwei Chen <yuwei.chen@intel.com>
Date: Wed, 29 Jul 2020 16:13:26 +0800
Subject: [PATCH] Test case12 for Add same module with different File Guid

Description: Add TestPkg/Sec/SecMain.inf in the dsc component with different File Guid.

Repo: edk2
Signed-off-by: Yuwei Chen <yuwei.chen@intel.com>
---
 TestPkg/LibraryClass.dsc | 2 ++
 TestPkg/Sec/SecMain.inf  | 1 +
 TestPkg/TestPkg32X64.dsc | 7 ++++++-
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/TestPkg/LibraryClass.dsc b/TestPkg/LibraryClass.dsc
index 6530bafbe7..936ec144df 100644
--- a/TestPkg/LibraryClass.dsc
+++ b/TestPkg/LibraryClass.dsc
@@ -44,10 +44,12 @@ NvVarsFileLib|TestPkg/Library/NvVarsFileLib/NvVarsFileLib.inf
   MtrrLib|UefiCpuPkg/Library/MtrrLib/MtrrLib.inf
   FakeLibraryLib|TestPkg/Library/BaseFakeLibraryLib/BaseFakeLibraryLib.inf
   LockBoxLib|MdeModulePkg\Library\LockBoxNullLib\LockBoxNullLib.inf
+  MemoryAllocationLib|MdeModulePkg\Library\BaseMemoryAllocationLibNull\BaseMemoryAllocationLibNull.inf
 
 [LibraryClasses.common.SEC]
   TimerLib|TestPkg/Library/AcpiTimerLib/BaseRomAcpiTimerLib.inf
   PpiListLib|EmulatorPkg/Library/SecPpiListLib/SecPpiListLib.inf
+  SortLib|MdeModulePkg\Library\BaseSortLib\BaseSortLib.inf
   
 !ifdef $(DEBUG_ON_SERIAL_PORT)
   DebugLib|MdePkg/Library/BaseDebugLibSerialPort/BaseDebugLibSerialPort.inf
diff --git a/TestPkg/Sec/SecMain.inf b/TestPkg/Sec/SecMain.inf
index 3066e0d102..a8f073f581 100644
--- a/TestPkg/Sec/SecMain.inf
+++ b/TestPkg/Sec/SecMain.inf
@@ -52,6 +52,7 @@
   LocalApicLib
   FakeLibraryLib
   LockBoxLib
+  SortLib
 
 [Ppis]
   gEfiTemporaryRamSupportPpiGuid                # PPI ALWAYS_PRODUCED
diff --git a/TestPkg/TestPkg32X64.dsc b/TestPkg/TestPkg32X64.dsc
index b24f19fc5f..f020a0ef18 100644
--- a/TestPkg/TestPkg32X64.dsc
+++ b/TestPkg/TestPkg32X64.dsc
@@ -119,7 +119,12 @@
     NULL|MdeModulePkg/Library/LzmaCustomDecompressLib/LzmaCustomDecompressLib.inf
 	LockBoxLib|MdeModulePkg\Library\LockBoxNullLib\LockBoxNullLib.inf
   }
-
+  TestPkg/Sec/SecMain.inf {
+    <Defines>
+	FILE_GUID = df1cfaf6-f381-4e64-9871-fc6a30fcc890
+	<LibraryClasses>
+    SortLib|MdeModulePkg\Library\BaseSortLib\BaseSortLib.inf
+  }
   #
   # PEI Phase modules
   #
-- 
2.27.0.windows.1

